VelocidiSDK/VelocidiSDK/VSDKVelodici.m

```diff
@@ -75,6 +85,22 @@ - (void)match: (NSString *)providerId
     onSuccess: (void (^)(NSURLResponse *response, id responseObject))onSuccessBlock
     onFailure: (void (^)(NSError * error))onFailureBlock {
     
+    if ([providerId length] <= 0){
+        NSException* ex = [NSException
+                exceptionWithName:@"InvalidArgument"
+                reason:@"Provider must not be empty!"
+                userInfo:nil];
+        @throw ex;
+    }
+    
+    if ([userIds count] < 2){
+        NSException* ex = [NSException
+                exceptionWithName:@"InvalidArgument"
+                reason:@"At least 2 user ids must be provided!"
+                userInfo:nil];
+        @throw ex;
+    }
+    
```

Comment:
I don't think we should be throwing exceptions here. We should return this error via the `onFailureBlock`

Reply:
Also, given that this is internal logic of how the match works, should we be doing this in `VSDKMatchRequest`?

Reply:
In 96af9d1 I try to refactor things to avoid throwing exceptions.